todo-app/src/TextField.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+
+export default class TextField extends React.Component {
+
+    render() {
+        if (this.props.multiline) {
+            return this.renderMultiLine();
+        } else if (this.props.singleline) { 
+            return this.renderSingleLine();
+        } else {
+            return this.renderDateForm();
```

Comment:
Your component is called `TextField`, but you've made it so that it also allows entering dates. It makes more sense to create a new component called `DateField` and use that instead. Now it's sort of confusing that if I use `<TextField/>` it's actually for a date, and if I use `<TextField singleline/>` it's a text field.